pkg/state/state.go

```diff
@@ -0,0 +1,59 @@
+package state
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+
+	"github.com/redhat-developer/odo/pkg/api"
+	"github.com/redhat-developer/odo/pkg/testingutil/filesystem"
+)
+
+type State struct {
+	content             Content
+	fs                  filesystem.Filesystem
+	getSecondsFromEpoch func() int64
+	getpid              func() int
+}
+
+func NewStateClient(fs filesystem.Filesystem) *State {
+	return &State{
+		fs:                  fs,
+		getSecondsFromEpoch: getSecondsFromEpoch,
+		getpid:              os.Getpid,
+	}
+}
+
+func (o *State) SetForwardedPorts(fwPorts []api.ForwardedPort) error {
+	// TODO(feloy) When other data is persisted into the state file, it will be needed to read the file first
+	o.content.ForwardedPorts = fwPorts
+	o.setMetadata()
```

Comment:
I like the idea of persisting such metadata as well. But wouldn't it make sense to persist the `PID` earlier, as soon as the command starts? I am thinking of scenarios where it could be interesting to get access to information like the PID as soon as possible, for example if the command is hanging before this `SetForwardedPorts` method is called.

Reply:
Also, I suggest changing the `timestamp` field with two different fields, to make the intent clearer. Because I think the meaning was not clear without looking at the PR description. Maybe fields like `createdAt` and `lastUpdatedAt`.

Reply:
That makes sense. I would prefer to discuss this during cabal to have a consensus before going further.

Reply:
Just for reference, it has been decided to remove both the timestamp and the PID for now. They may be added later on if needed.